provider/uprops/src/provider.rs

```diff
@@ -2,57 +2,38 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use crate::bin_uniset::BinaryPropertyUnicodeSetDataProvider;
-use crate::enum_uniset::EnumeratedPropertyUnicodeSetDataProvider;
-use crate::uprops_helpers::get_last_component_no_version;
-use icu_properties::provider::UnicodePropertyV1Marker;
+use icu_provider::fork::by_key::MultiForkByKeyProvider;
 use icu_provider::iter::IterableDynProvider;
-use icu_provider::prelude::*;
-
+use icu_provider::serde::SerializeMarker;
 use std::path::Path;
 
 /// This data provider returns `UnicodeSet` data inside a `UnicodeProperty`
 /// data struct. The source data is in the form of a directory of TOML file(s)
 /// of data for the property(-ies) desired, as given by the ICU4C property data
 /// exporter tool.
-pub struct PropertiesDataProvider {
-    binary: BinaryPropertyUnicodeSetDataProvider,
-    enumerated: EnumeratedPropertyUnicodeSetDataProvider,
-}
-
+pub struct PropertiesDataProvider;
```

Comment:
Nit: Given a choice between naked function and fake struct, I prefer naked function.  Naming ideas:

- `create_exportable_properties_provider<T>()`
- `create_exportable_serialize_properties_provider()`

Can you return:

`eyre::Result<MultiForkByKeyProvider<Box<dyn IterableDynProvider<T>>>>`

and then parameterize the function using a `where` to make `T` valid?